read-fonts/src/tables/gvar.rs

```diff
@@ -0,0 +1,489 @@
+        let data = header.serialized_data()?;
+
+        // if there are shared point numbers, get them now
+        let (shared_point_numbers, serialized_data) =
+            if header.tuple_variation_count().shared_point_numbers() {
+                let (packed, data) = PackedPointNumbers::split_off_front(data);
+                (Some(packed), data)
+            } else {
+                (None, data)
+            };
+
+        Ok(GlyphVariationData {
+            tuple_count: count,
+            axis_count,
+            shared_tuples,
+            shared_point_numbers,
+            header_data,
+            serialized_data,
+        })
+    }
+
+    /// Return an iterator over all of the variation tuples for this glyph.
+    pub fn tuples(&self) -> TupleVariationIter<'a> {
+        TupleVariationIter {
+            current: 0,
+            parent: self.clone(),
+            header_iter: TupleVariationHeaderIter::new(
+                self.header_data,
+                self.tuple_count(),
+                self.axis_count,
+            ),
+            serialized_data: self.serialized_data,
+        }
+    }
+
+    fn tuple_count(&self) -> usize {
+        self.tuple_count.count() as usize
+    }
+}
+
+/// An iterator over the [`TupleVariation`]s for a specific glyph.
+pub struct TupleVariationIter<'a> {
+    current: usize,
+    parent: GlyphVariationData<'a>,
+    header_iter: TupleVariationHeaderIter<'a>,
+    serialized_data: FontData<'a>,
+}
+
+impl<'a> TupleVariationIter<'a> {
+    fn next_tuple(&mut self) -> Option<TupleVariation<'a>> {
+        if self.parent.tuple_count() == self.current {
+            return None;
+        }
+        self.current += 1;
+
+        // FIXME: is it okay to discard an error here?
+        let header = self.header_iter.next()?.ok()?;
+        let data_len = header.variation_data_size() as usize;
+        let var_data = self.serialized_data.take_up_to(data_len)?;
+
+        let (point_numbers, packed_deltas) = if header.tuple_index().private_point_numbers() {
+            PackedPointNumbers::split_off_front(var_data)
+        } else {
+            (self.parent.shared_point_numbers.clone()?, var_data)
+        };
+        Some(TupleVariation {
+            axis_count: self.parent.axis_count,
+            header,
+            shared_tuples: self.parent.shared_tuples.clone(),
+            packed_deltas: PackedDeltas::new(packed_deltas),
+            point_numbers,
+        })
+    }
+}
+
+impl<'a> Iterator for TupleVariationIter<'a> {
+    type Item = TupleVariation<'a>;
+
+    fn next(&mut self) -> Option<Self::Item> {
+        self.next_tuple()
+    }
+}
+
+/// A single set of tuple variation data
+#[derive(Clone)]
+pub struct TupleVariation<'a> {
+    axis_count: u16,
+    header: TupleVariationHeader<'a>,
+    shared_tuples: SharedTuples<'a>,
+    packed_deltas: PackedDeltas<'a>,
+    point_numbers: PackedPointNumbers<'a>,
+}
+
+impl<'a> TupleVariation<'a> {
```

Comment:
```suggestion
impl<'a> TupleVariation<'a> {
    /// Returns true if this tuple provides deltas for all points in a glyph.
    pub fn all_points(&self) -> bool {
        self.point_numbers.count() == 0
    }
    
```
Add a method to detect if a tuple provides deltas for all points. This allows us to skip the interpolation code path for these tuples which is a pretty big performance win.